gulpfile.js

```diff
@@ -3,40 +3,75 @@ var deployDir = 'docs';
 
 var gulp = require('gulp');
 var clean = require('gulp-clean');
+var concat = require('gulp-concat');
 var jshint = require('gulp-jshint');
+var uglify = require('gulp-uglify');
+var babel = require('gulp-babel');
+var inject = require('gulp-inject');
+var minifyHtml = require('gulp-minify-html');
+var minifyCss = require('gulp-clean-css');
+var browserSync = require('browser-sync');
+var injectVersion = require('gulp-inject-version');
 
-gulp.task('default', ['copy']);
+gulp.task('default', ['lint', 'html']);
 
-gulp.task('copy', ['clean', 'lint'], function() {
-    return gulp.src(src)
-        .pipe(gulp.dest(deployDir));
+gulp.task('build', ['build-html']);
+
+gulp.task('clean', function () {
+  return gulp.src(deployDir)
+      .pipe(clean());
+});
+
+gulp.task('lint', function () {
+  var filesToLint = src.map(path => path + '/*.js');
+  filesToLint.push('gulpfile.js');
+
+  return gulp.src(filesToLint)
+      .pipe(jshint({
+        eqeqeq: true,
+        esversion: 6,
+        eqnull: true
+      }))
+      .pipe(jshint.reporter('jshint-stylish'))
+      .pipe(jshint.reporter('fail'));
 });
 
-gulp.task('clean', function() {
-    return gulp.src(deployDir)
-        .pipe(clean());
+gulp.task('build-js', ['clean'], function () {
+  return gulp.src(src.map(path => path + '/*.js'))
+      .pipe(injectVersion())
+      .pipe(babel({
+        presets: ['es2015']
+      }))
+      .pipe(concat('all.min.js'))
+      .pipe(uglify())
+      .pipe(gulp.dest(deployDir));
 });
 
-gulp.task('lint', function() {
-    return gulp.src(src.map(function(path) {
-        return path+'/*.js';
-    }))
-        .pipe(jshint({
-            eqeqeq: true,
-            esversion: 6,
-            eqnull: true
-        }))
-        .pipe(jshint.reporter('jshint-stylish'))
-        .pipe(jshint.reporter('fail'));
+gulp.task('build-css', ['clean'], function () {
+  return gulp.src('src/**/*.css')
+      .pipe(minifyCss())
+      .pipe(gulp.dest(deployDir));
+});
+
+gulp.task('build-html', ['build-js', 'build-css'], function () {
+  var target = gulp.src('src/index.html');
+  var sources = gulp.src(deployDir + '/*.{js,css}');
```

Comment:
`gulp-inject` only injects the file references (script and link tags) not the entire file. All the files in the `docs/` directory are important.

The JS and CSS files in `docs/` don't get modified at this point; it only gets a list of the files paths (I'll add `{read: false}` to the `gulp.src` to enforce that).

And I would think that if/when we bring in third-party libraries and components, we can update the gulpfile to make it easier to work with.